benchmarks/dataframe.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+import time, argparse
+import numpy as np
+import pandas as pd
+
+import arkouda as ak
+
+OPS = ['_get_head_tail_server', '_get_head_tail']
+TYPES = ('int64', 'uint64',)
+
+def generate_dataframe(N):
+    types = [ak.Categorical, ak.pdarray, ak.Strings, ak.SegArray]
+
+    # generate random columns to build dataframe
+    df_dict = {}
+    for x in range(20):  # loop to create 20 random columns
+        key = f"c_{x}"
+        d = types[x % 4]
+        if d == ak.Categorical:
+            str_arr = ak.random_strings_uniform(minlen=5, maxlen=6, size=N)
+            df_dict[key] = ak.Categorical(str_arr)
+        elif d == ak.pdarray:
+            df_dict[key] = ak.array(np.random.randint(0, 2 ** 32, N))
+        elif d == ak.Strings:
+            df_dict[key] = ak.random_strings_uniform(minlen=5, maxlen=6, size=N)
+        elif d == ak.SegArray:
+            df_dict[key] = ak.SegArray(ak.arange(0, N*5, 5), ak.array(np.random.randint(0, 2 ** 32, N*5)))
+
+    return ak.DataFrame(df_dict)
+
+def time_ak_df_display(N_per_locale, trials):
+    print(">>> arkouda dataframe display")
+    cfg = ak.get_config()
+    N = N_per_locale * cfg["numLocales"]
+    print("numLocales = {}, N = {:,}".format(cfg["numLocales"], N))
+
+    pd.set_option("display.max_rows", 100)
+    pd.set_option("display.min_rows", 10)
+    pd.set_option("display.max_columns", 20)
+
+    df = generate_dataframe(N)
+
+    timings = {op: [] for op in OPS}
+    results = {}
+    for i in range(trials):
+        timings = {op: [] for op in OPS}
+        for op in timings.keys():
+            fxn = getattr(df, op)
+            start = time.time()
+            r = fxn()
+            end = time.time()
+            timings[op].append(end - start)
+            results[op] = r
+
+    tavg = {op: sum(t) / trials for op, t in timings.items()}
+
+    for op, t in tavg.items():
+        print("  {} Average time = {:.4f} sec".format(op, t))
+        bytes_per_sec = (df.size * 64 * 2) / t
+        print("  {} Average rate = {:.2f} GiB/sec".format(op, bytes_per_sec / 2 ** 30))
+
+def check_correctness(N_per_locale):
+    cfg = ak.get_config()
+    N = N_per_locale * cfg["numLocales"]
+    df = generate_dataframe(N)
+
+    pd.set_option("display.max_rows", 100)
+    pd.set_option("display.min_rows", 10)
+    pd.set_option("display.max_columns", 20)
+
+    printdf = df._get_head_tail_server() # measure the pandas df returned
+    # Mainly want to verify shape for the print
+    assert(printdf.shape[0] == 101)
+    assert(printdf.shape[1] == 20)
+
+
+def create_parser():
+    parser = argparse.ArgumentParser(description="Run the setops benchmarks: intersect1d, union1d, setdiff1d, setxor1d")
+    parser.add_argument('hostname', help='Hostname of arkouda server')
+    parser.add_argument('port', type=int, help='Port of arkouda server')
+    parser.add_argument('-n', '--size', type=int, default=10**4, help='Problem size: length of arrays A and B')
```

Comment:
I'm not sure but `--size` might need updating as well

Reply:
Updated the description.